api/options.go

```diff
@@ -0,0 +1,110 @@
+package api
+
+import (
+	"context"
+	"net/http"
+
+	dperrors "github.com/ONSdigital/dp-cantabular-filter-flex-api/errors"
+	"github.com/ONSdigital/dp-cantabular-filter-flex-api/model"
+	"github.com/go-chi/chi/v5"
+	"github.com/pkg/errors"
+)
+
+func (api *API) getFilterDimensionOptions(w http.ResponseWriter, r *http.Request) {
+	ctx := context.Background()
+
+	filterID := chi.URLParam(r, "id")
+	dimensionName := chi.URLParam(r, "name")
+
+	pageLimit, offset, err := getPaginationParams(r.URL, api.cfg.DefaultMaximumLimit)
+	if err != nil {
+		api.respond.Error(
+			ctx,
+			w,
+			http.StatusBadRequest,
+			errors.Wrap(err, "Bad Request: "),
+		)
+
+	}
+
+	if pageLimit == 0 {
+		// define a reasonable default
+		// in light of bad input
+		// also slice not work with 0
+		pageLimit = 10
+	}
+
+	var eTag string
+	if reqETag := api.getETag(r); reqETag != eTagAny {
+		eTag = reqETag
+	}
+
+	options, totalCount, err := api.store.GetFilterDimensionOptions(
+		ctx,
+		filterID,
+		dimensionName,
+		pageLimit,
+		offset,
+	)
+	if err != nil {
+		status := statusCode(err)
+		finalErr := errors.New("internal server error")
+
+		if dperrors.NotFound(err) {
+			status = http.StatusNotFound
```

Comment:
I'm not sure what the point in this is.. These will assert to these status codes automatically when using statusCode(err) in the call to respond.Error()

Reply:
If the filter is not found, then the spec says return a 404. If the dimension name is not found within a filter, then says to return a 400 bad request. So in the query that I wrote, there are two ways in which it could fail. The other way it could fail is connection errors. This was the most terse but readable way I could think of to handle the different types of errors coming from the function call

Reply:
The point I'm getting at is, under the hood respond.Error() does exactly the same thing under the hood, so this is redundant. Passing an error that will assert to dperrors.NotFound == true will automatically set the status code to 404.

Reply:
As discussed, and with a better understanding of `StatusCode(err)`, I have reworked the code to remove manual checks, lmk what you think